### PR TITLE
Expose renderDate function for DatePicker

### PR DESCRIPTION
**Pull request by henrinormak**
Exposed [rc-calendar's](https://github.com/react-component/calendar) function for rendering individual date cells in the calendar. This can be used to customise certain dates in `DatePicker` and `RangePicker` (such as the first of the month in the added demo).

This is currently missing documentation in Chinese, I was hoping you could help me out with that (I don't unfortunately speak it myself).

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [x] Add unit tests for the feature.

Close: #7216

### CODE REVIEW

**Comment by henrinormak** [comment]
I am not entirely sure I chose the correct branch for this PR, I couldn't see any feature branches...

**Comment by afc163** [comment]
`antd-3.0` is a correct branch cause we have locked the 2.x from new features.

**Comment by henrinormak** [comment]
Redirected and rebased against the correct branch. I think Travis is failing because of an error in FormItem that is fixed in #7654.

**Comment by afc163** [comment]
6ad3fe5

Try rebase again.

**Comment by henrinormak** [comment]
Also, forgot to ask, when is the 3.0 release scheduled to happen? Is there a date set? I am trying to plan my work (which depends on this PR) and would if I could better understand the timeline.

File: components/date-picker/RangePicker.tsx [hunk 1]
```diff
@@ -179,6 +179,7 @@ export default class RangePicker extends React.Component<any, any> {
       disabledDate, disabledTime,
       showTime, showToday,
       ranges, onOk, locale, format,
+      renderDate,
```
Comment (benjycui) [inline components/date-picker/RangePicker.tsx]: Just use `dateRender`, don't rename it..

**Comment by benjycui** [comment]
Close: https://github.com/ant-design/ant-design/issues/7216

**Comment by henrinormak** [comment]
Reverted to use the original name. I was trying to avoid the inconsistency (present in rc-calendar) where different rendering related functions don't have a similar naming convention - for example `renderExtraFooter`, but I suppose it doesn't really matter.

**Review by benjycui — APPROVED** [review]

**Pull request merged**